autosklearn/metalearning/metafeatures/metafeatures.py

```diff
@@ -235,7 +235,10 @@ def _calculate(self, X, y, logger, categorical):
 @metafeatures.define("NumberOfMissingValues", dependency="MissingValues")
 class NumberOfMissingValues(MetaFeature):
     def _calculate(self, X, y, logger, categorical):
-        return float(helper_functions.get_value("MissingValues").sum())
+        if scipy.sparse.issparse(X):
+            return float(helper_functions.get_value("MissingValues").sum())
+        else:
+            return float(np.count_nonzero(helper_functions.get_value("MissingValues")))
```

Comment:
Yes exactly, for dataframes it returns a list per column.

Reply:
I will mark this as resolved, not only because unit testing seem to pass with this, but also it seems to provide the same values on the datasets that I tried (in numpy and pandas). I also moved the `np.finite` to the `pd.isna/pd.na` equivalent, as for dataframes if you have a category, numpy does not support this.